staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go

```diff
@@ -196,22 +197,15 @@ func (t *envelopeTransformer) addTransformer(encKey []byte, key []byte) (value.T
 	transformer := t.baseTransformerFunc(block)
 	// Use base64 of encKey as the key into the cache because hashicorp/golang-lru
 	// cannot hash []uint8.
-	if t.cacheEnabled {
-		t.transformers.Add(base64.StdEncoding.EncodeToString(encKey), transformer)
-		metrics.RecordDekCacheFillPercent(float64(t.transformers.Len()) / float64(t.cacheSize))
-	}
+	t.cache.set(base64.StdEncoding.EncodeToString(encKey), &cacheRecord{transformer: transformer}, t.cacheTTL)
```

Comment:
I would move both the base64 and cache record bits into the cache and let the caller pass in the encryption key bytes and the transformer

```go
set(key []bytes, transformer value.Transformer)
```